account-management-app/src/main/webapp/WEB-INF/jspx/includes/storage-provider-form.jspx

```diff
@@ -1,190 +1,241 @@
                     <form:select
-                     cssErrorClass="error"
-                     id="${param.storageProvider}.properties['AWS_REGION']"
-                     path="${param.storageProvider}.properties['AWS_REGION']"
-                     onclick="same(this.form)">
-                       <form:option value="" label="" />
-                       <c:forEach items="${awsregions}" var="region">
-                         <form:option value="${region.name}" label="${region.name}" />
-                       </c:forEach>
+                        cssErrorClass="error"
+                        id="${param.storageProvider}.properties['AWS_REGION']"
+                        path="${param.storageProvider}.properties['AWS_REGION']"
+                        onclick="same(this.form)">
+                        <form:option value="" label="" />
+                        <c:forEach items="${awsregions}" var="region">
+                            <form:option value="${region.name}" label="${region.name}" />
+                        </c:forEach>
                     </form:select>
-	                <form:errors
-	                 path="${param.storageProvider}.properties['AWS_REGION']"
-	                 cssClass="error"
-	                 element="div" />
-	              </li>
-	            </c:if>
+                    <form:errors
+                        path="${param.storageProvider}.properties['AWS_REGION']"
+                        cssClass="error"
+                        element="div" />
+                </li>
+            </c:if>
+            <c:if test="${param.storageProviderType eq 'SWIFT_S3'}">
+                <li>
+                    <form:label
+                        cssErrorClass="error"
+                        path="${param.storageProvider}.properties['S3_ENDPOINT']">
+                        <spring:message code="s3.endpoint"/>
+                    </form:label>
+                    <form:input
+                        cssErrorClass="error"
+                        path="${param.storageProvider}.properties['S3_ENDPOINT']"
+                        onclick="same(this.form)"/>
+                    <form:errors
+                        path="${param.storageProvider}.properties['S3_ENDPOINT']"
+                        cssClass="error"
+                        element="div" />
+                </li>
+                <li>
+                    <form:label
+                        cssErrorClass="error"
+                        path="${param.storageProvider}.properties['S3_SIGNER_TYPE']">
+                        <spring:message code="s3.signer.type"/>
+                    </form:label>
+                    <form:input
+                        cssErrorClass="error"
+                        path="${param.storageProvider}.properties['S3_SIGNER_TYPE']"
+                        onclick="same(this.form)"/>
+                    <form:errors
+                        path="${param.storageProvider}.properties['S3_SIGNER_TYPE']"
```

Comment:
Same note about naming as above - whatever changes are made on the DuraCloud end will need to be made here as well.